src/dotnet/Audio.UI.Blazor/Components/AudioRecorder/workers/opus-encoder-worker.ts

```diff
@@ -152,6 +156,18 @@ async function onInit(message: InitEncoderMessage): Promise<void> {
         encoder.encode(lowNoiseChunk.buffer);
     }
 
+    if (whenMicReady) {
+        // wait for mic data
+        await whenMicReady;
+        whenMicReady = null;
```

Comment:
Why do you reset to null whenMicReady? onInit method is supposed to be called more than once?